### PR TITLE
Add proper condition for installing uvloop

### LINKED ISSUE #594: ModuleNotFoundError: No module named 'uvloop'
### Bug Description

When trying to work through the Getting Started example, I get the following error when trying to run the service:

```
Traceback (most recent call last):
  File "/Users/jason/src/python/robyn-app/venv/lib/python3.11/site-packages/multiprocess/process.py", line 314, in _bootstrap
    self.run()
  File "/Users/jason/src/python/robyn-app/venv/lib/python3.11/site-packages/multiprocess/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/jason/src/python/robyn-app/venv/lib/python3.11/site-packages/robyn/processpool.py", line 162, in spawn_process
    loop = initialize_event_loop()
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jason/src/python/robyn-app/venv/lib/python3.11/site-packages/robyn/processpool.py", line 127, in initialize_event_loop
    import uvloop
ModuleNotFoundError: No module named 'uvloop'
```

### Steps to Reproduce

1. Work through [getting started](https://sparckles.github.io/robyn/#/getting-started)

### Your operating system

MacOS 13.4.1 (intel processor)

### Your Python version (`python --version`)

3.11.3

### Your Robyn version

0.38.0

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by banipreetr**
**Description**

This PR fixes #594

### CODE REVIEW

**Comment by banipreetr** [comment]
@sansyrox 

Can you please approve the rest of the workflows in CI ? 
image

**Comment by sansyrox** [comment]
@banipreetr done 😄

File: pyproject.toml [hunk 1]
```diff
@@ -30,7 +30,7 @@ dependencies = [
   'nestd == 0.3.1',
   'inquirerpy == 0.3.4',
 # conditional
-  "uvloop>=0.17.0; sys_platform == 'darwin' and platform_machine == 'aarch_64' and platform_machine == 'x86_64' and platform_machine == 'i686'"
+  "uvloop>=0.17.0; sys_platform == 'darwin' and (platform_machine == 'aarch_64' or platform_machine == 'x86_64' or platform_machine == 'i686')"
```
[the remark this answers is not in the document]
Comment (sansyrox) [reply pyproject.toml:33]: This condition will only install uvloop on osx and not other platforms.

Comment (sansyrox) [inline pyproject.toml:33]:
```suggestion
  "uvloop~=0.17.0; sys_platform != 'win32' and platform_python_implementation == 'CPython'"
```

File: pyproject.toml [hunk 2]
```diff
@@ -30,7 +30,7 @@ dependencies = [
   'nestd == 0.3.1',
   'inquirerpy == 0.3.4',
 # conditional
-  "uvloop>=0.17.0; sys_platform == 'darwin' and platform_machine == 'aarch_64' and platform_machine == 'x86_64' and platform_machine == 'i686'"
+  "uvloop~=0.17.0; sys_platform != 'win32' and platform_python_implementation == 'CPython'"
```
Comment (banipreetr) [inline pyproject.toml:33]: Hey @sansyrox ! Thanks for pushing this. Can you please explain the condition:

```
platform_python_implementation == 'CPython'
```

Thanks.

Comment (sansyrox) [reply pyproject.toml:33]: Hey @banipreetr 👋 

This is needed when the users try to use robyn with `PyPy`. While we don't explicitly show support for PyPy on our repo. I am 70% sure that robyn works with PyPy lol

Comment (sansyrox) [inline pyproject.toml:33]:
```suggestion
uvloop~=0.17.0; sys_platform != 'win32' and platform_python_implementation == 'CPython' and platform_machine != 'armv7l'
```

File: pyproject.toml [hunk 3]
```diff
@@ -30,7 +30,7 @@ dependencies = [
   'nestd == 0.3.1',
   'inquirerpy == 0.3.4',
 # conditional
-  "uvloop>=0.17.0; sys_platform == 'darwin' and platform_machine == 'aarch_64' and platform_machine == 'x86_64' and platform_machine == 'i686'"
+uvloop~=0.17.0; sys_platform != 'win32' and platform_python_implementation == 'CPython' and platform_machine != 'armv7l'
```
Comment (sansyrox) [inline pyproject.toml:33]:
```suggestion
  "uvloop~=0.17.0; sys_platform != 'win32' and platform_python_implementation == 'CPython' and platform_machine != 'armv7l'"
```

**Review by sansyrox — APPROVED** [review]
LGTM! :fire:

**Pull request merged**